.github/ci.yml

```diff
@@ -0,0 +1,85 @@
+name: CI
+
+env:
+  JDK_JAVA_OPTIONS: -XX:+PrintCommandLineFlags # JDK_JAVA_OPTIONS is _the_ env. variable to use for modern Java
+  JVM_OPTS: -XX:+PrintCommandLineFlags # for Java 8 only (sadly, it is not modern enough for JDK_JAVA_OPTIONS)
+
+on:
+  pull_request:
+  push:
+    branches: ['master']
+  release:
+    types:
+      - published
+
+jobs:
+  lint:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 30
+    steps:
+      - name: Checkout current branch
+        uses: actions/checkout@v2.4.0
+        with:
+          fetch-depth: 0
+      - name: Setup Scala and Java
+        uses: olafurpg/setup-scala@v10
+      - name: Cache scala dependencies
+        uses: coursier/cache-action@v6
+      - name: Lint code
+        run: sbt ++3.1.3 check
+
+  test:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 30
+    strategy:
+      fail-fast: false
+      matrix:
+        java: ['adopt@1.8', 'adopt@1.11']
+        scala: ['3.1.3']
+        platform: ['JVM', 'JS']
+    steps:
+      - name: Checkout current branch
+        uses: actions/checkout@v2.4.0
+        with:
+          fetch-depth: 0
+      - name: Setup Scala and Java
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: ${{ matrix.java }}
+      - name: Cache scala dependencies
+        uses: coursier/cache-action@v6
+      - name: Run tests
+        if: ${{ !startsWith(matrix.scala, '3.1.') }}
+        run: sbt ++${{ matrix.scala }}! test${{ matrix.platform }}
+      - name: Run dotty tests
+        if: ${{ startsWith(matrix.scala, '3.1.') && matrix.platform == 'JVM' }}
+        run: sbt ++${{ matrix.scala }}! testJVM
+
+  ci:
+    runs-on: ubuntu-20.04
+    needs: [lint, test]
+    steps:
+      - name: Aggregate of lint, and all tests
+        run: echo "ci passed"
+
+  publish:
```

Comment:
I think I'd prefer to not automatically publish anything.
It's fine if the GA run linting and tests but no automatic.
A proper release requires several higher-level tasks like updating release notes, documentation and version numbers that cannot be properly automated.

Reply:
That is fine, I wanted you to have the opportunity to choose, and this PR is our best channel for communication.

Reply:
Yes, that's very good! Thank you!